includes/YesWiki.php

```diff
@@ -1402,6 +1398,39 @@ public function loadExtensions()
         $this->config['templates'] = $this->services->get(ThemeManager::class)->loadTemplates($metadata);
     }
 
+    /**
+     * Replace recursively all the indexed arrays of $array1 with the corresponding indexed array of $array2
+     * @param $array1 the first array that is merged
+     * @param $array2 the second array that give the value for indexed array
+     */
+    public function replaceRecursivelyIndexedArrays(&$array1, &$array2) {
+        foreach ($array2 as $key => $val) {
+            if (is_array($val)) {
+                if(!$this->isAssocArray($val)) {
+                    if($array1[$key] != $val) {
+                        $array1[$key] = $val;
+                    }
+                } else {
+                    $subarray1 = &$array1[$key];
+                    $subarray2 = &$array2[$key];
+                    $this->replaceRecursivelyIndexedArrays($subarray1, $subarray2);
```

Comment:
```suggestion
                    $this->replaceRecursivelyIndexedArrays($subarray1, $subarray2);
```
Je ne sais pas si le comportement attendu et de le faire récursivement.
En effet, ça me parait logique pour le premier niveau, mais pour les niveaux suivants, je me demande si ça n'est pas mieux de faire une fusion entre les deux. Est-ce que tu aurais une idée de cas d'exemple pour départager le choix ?

Reply:
Je te confirme que c'est logique qu'il y ait le même comportement pour tous les tableaux du fichier de config. Par exemple, si je veux redéfinir un tableau au niveau du module lms (tableau associatif `lms_config`), je ne vois pas pourquoi on ne s'attendrait pas au même comportement.
Là ça touche à une autre question de compartimenter ou non les configurations des extensions au lieu d'avoir un préfixe de variable comme c'est souvent le cas (notamment pour bazar).

Mais si on prend un exemple d'une autre extension qui n'utilise pas ce compartimentage, on a également ce besoin de pouvoir redéfinir un tableau : 
```
'yeswiki-farm-extra-config' => [
    'meta' => [
      'robots' => 'noindex, nofollow',
    ],
    ...
]
```

Est-ce que c'est plus clair pour toi ?

Reply:
Oui nickel. Merci pour la confirmation. Je fusionne donc car je crois que nous avons tous les deux fait des commits et validé ceux de l'autre respectivement.